packages/fxa-settings/src/components/LinkRememberPassword/index.tsx

```diff
@@ -5,31 +5,33 @@
 import React from 'react';
 import { FtlMsg } from 'fxa-react/lib/utils';
 import { useLocation } from '@reach/router';
+import { isEmailValid } from 'fxa-shared/email/helpers';
 
 export type LinkRememberPasswordProps = {
   email?: string;
   forceAuth?: boolean;
 };
 
-const LinkRememberPassword = ({
-  email,
-  forceAuth = false,
-}: LinkRememberPasswordProps) => {
+const LinkRememberPassword = ({ email }: LinkRememberPasswordProps) => {
   const location = useLocation();
   const params = new URLSearchParams(location.search);
-  if (email) {
+  let linkHref: string;
+  if (email && isEmailValid(email)) {
     params.set('email', email);
+    linkHref = `/signin?${params}`;
+  } else {
+    linkHref = params.size > 0 ? `/?${params}` : '/';
```

Comment:
TIL `URLSearchParams().size`!

Out of scope here to update all the instances but we have to do this so much I feel like we should make a helper to do this for us so we don't have to think about it, we're doing `queryParams.length > 1 && queryParams) || ''` in at least one other spot...